compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -810,11 +810,35 @@ setReductionVectorDistributionConfig(IREE::GPU::TargetAttr target,
       threadLoads /= 2;
     }
   }
+
+  std::optional<int64_t> parallelSize = 1;
+  for (int64_t dim : parallelDims) {
+    if (ShapedType::isDynamic(bounds[dim])) {
+      parallelSize = std::nullopt;
+      break;
+    }
+    *parallelSize *= bounds[dim];
+  }
+
   // Deduce the workgroup size we should use for reduction. Currently a
   // workgroup processes all elements in reduction dimensions. Need to make sure
   // the workgroup size we use can divide the total reduction size, and it's
   // also within hardware limitations.
-  const int64_t maxWorkgroupSize = 1024;
+
+  ArrayRef<int32_t> maxWgSizes = wgp.getMaxWorkgroupSizes();
+  int32_t maxWorkgroupSize = *llvm::max_element(maxWgSizes);
+  IREE::GPU::TargetChipAttr chip = target.getChip();
+  const int numComputeUnits = chip ? chip.getWgpCount() : 256;
```

Comment:
```suggestion
  const int numWGPs = chip ? chip.getWgpCount() : 256;
```
Let's keep the naming consistent. WGP == CU